### DISCUSSION TITLE
How to customize the generated Markdown

### DISCUSSION THREAD

**Discussion by testandattack** [Doraku/DefaultDocumentation#114] [Q&A]
_Apologies if the answer is already documented and I am just missing it._

In the generated documentation for a particular method, I am getting two lines of text for each parameter (as seen below). I would like to be able to more clearly delineate between each parameter, either by changing the formatting of the parameter's name, or adding lines between each set, or something else (examples below the initial screenshot). Any thoughts on the easiest way to do this?

![image](https://user-images.githubusercontent.com/86054303/172625316-c2b94b78-8ae3-4018-bc4d-8505e64c1fc5.png)

### Example Idea 1:
- Changed param names from `code` to **bold**
- Removed extra line from between name and description
- Changed the "Parameters" and "Returns" headers to H3 (from H4)

![image](https://user-images.githubusercontent.com/86054303/172625560-7b64470f-b4b0-45d4-bdef-95a8f5186bfa.png)

### Example Idea 2:

- Removed extra line from between name and description
- Added a line between each section using `<hr width="95%" style="background-color: #000000;" />`

![image](https://user-images.githubusercontent.com/86054303/172625768-24f7fd40-9376-4baa-a285-f2611bd6c2c2.png)

**Comment by Doraku** [comment]
This can be achieved by creating a plugin with a custom section for the `ParameterDocItem` type. See the [plugin sample](https://github.com/Doraku/DefaultDocumentation/tree/master/source/Sample) for how to create one. Then your custom section would look something like this:
```csharp
public sealed class ParameterTitleSection : ISection
{
    // important to use a new name so your implementation does not override the default Title section for the other types
    public string Name => "ParameterTitle";

    public void Write(IWriter writer)
    {
        if (writer.GetCurrentItem() is ParameterDocItem docItem)
        {
            if (!docItem.HasOwnPage(writer.Context))
            {
                string url = writer.Context.GetUrl(docItem);
                int startIndex = url.IndexOf('#') + 1;
                writer
                    .EnsureLineStartAndAppendLine()
                    .Append("<a name='")
                    .Append(url.Substring(startIndex, url.Length - startIndex))
                    .Append("'></a>");
            }

            // this is where you can customize what is written
            writer
                .EnsureLineStartAndAppendLine();
                .Append($"`{docItem.Name}` ")
                .AppendLink(docItem, parameterItem.Parameter.Type)
        }
    }
}
```

All that's left then is to tell in your configuration to use this section in your configuration:
```json
{
    ...
    "ParameterDocItem": {
        "Sections": [
            "Header",
            "ParameterTitle",
            "Summary",
            "Example",
            "Remarks",
            "SeeAlso"
        ]
    }
}
```
